test/unit/output/pudltabl_test.py

```diff
@@ -0,0 +1,130 @@
+"""Unit tests for PudlTabl class methods."""
+
+from datetime import date, datetime
+
+import pandas as pd
+import pyarrow as pa
+import pytest
+
+from pudl.output.pudltabl import PudlTabl
+
+
+class TestPudlTablAggTableName:
+    """Test PudlTabl._agg_table_name method."""
+
+    def test_yearly_aggregation(self):
+        """Test table name substitution for yearly aggregation."""
+        pudl_tabl = PudlTabl(freq="YS")
+
+        result = pudl_tabl._agg_table_name("out_eia923__AGG_generation")
+        assert result == "out_eia923__yearly_generation"
+
+    def test_monthly_aggregation(self):
+        """Test table name substitution for monthly aggregation."""
+        pudl_tabl = PudlTabl(freq="MS")
+
+        result = pudl_tabl._agg_table_name("out_eia923__AGG_fuel_receipts")
+        assert result == "out_eia923__monthly_fuel_receipts"
+
+    def test_no_aggregation_removes_agg(self):
+        """Test that _AGG is removed when freq is None."""
+        pudl_tabl = PudlTabl(freq=None)
+
+        result = pudl_tabl._agg_table_name("out_eia923__AGG_generation")
+        assert result == "out_eia923__generation"
+
+    def test_non_aggregated_table_unchanged(self):
+        """Test that non-aggregated table names are unchanged."""
+        pudl_tabl = PudlTabl(freq="YS")
+
+        result = pudl_tabl._agg_table_name("core_eia860__scd_generators")
+        assert result == "core_eia860__scd_generators"
+
+
+class TestPudlTablInitialization:
+    """Test PudlTabl initialization and validation."""
+
+    def test_valid_frequencies(self):
+        """Test that valid frequencies are accepted."""
+        # These should all work without raising exceptions
+        PudlTabl(freq=None)
+        PudlTabl(freq="YS")
+        PudlTabl(freq="MS")
+
+    def test_invalid_frequency_raises_error(self):
+        """Test that invalid frequencies raise ValueError."""
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="QS")  # Quarterly not supported
+
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="D")  # Daily not supported
+
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="invalid")
+
+    def test_date_conversion(self):
```

Comment:
Similarly to above, this seems to mostly test some pretty trivial code: "does `pd.to_datetime` work?" - here are some other related things that might be more useful to test:

* "do start/end default to the right dates?" (though that might live in a pudl helpers test instead of pudltabl test)
* given start/end dates (or a lack of them), do we get the expected filter behavior?